velox/functions/prestosql/Map.cpp

```diff
@@ -77,6 +77,95 @@ class MapFunction : public exec::VectorFunction {
         checkDuplicateKeys(mapVector, rows, context);
       }
       context.moveOrCopyResult(mapVector, rows, result);
+    } else if (decodedKeys->isConstantMapping()) {
+      // Constant keys.
+      auto keysIndex = decodedKeys->index(rows.begin());
+      auto valueIndices = decodedValues->indices();
+
+      auto keysArray = decodedKeys->base()->as<ArrayVector>();
+      auto valuesArray = decodedValues->base()->as<ArrayVector>();
+
+      // Verify there are no null keys and no duplicate keys.
+      auto numKeys = keysArray->sizeAt(keysIndex);
+      auto keysElements = keysArray->elements();
+      auto keysOffset = keysArray->offsetAt(keysIndex);
+
+      try {
+        if (keysElements->mayHaveNulls()) {
+          for (auto i = 0; i < numKeys; ++i) {
+            VELOX_USER_CHECK(!keysElements->isNullAt(keysOffset + i), kNullKey);
```

Comment:
should we add a more descriptive user facing error here, like "found null on map keys" or something similar?

Reply:
That's a good point. Presto uses 'map key cannot be null'. Let me use that.

```
presto:di> select map(array[null], array[1]);
Query 20230510_105602_19150_4466g failed: map key cannot be null
```

Reply:
Actually, this is already the message:

```
static const char* kNullKey = "map key cannot be null";
```